File: Data Structures & Algorithms/three-integer-sum/submission-1.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        
        List<List<Integer>> res = new ArrayList <>();
        Arrays.sort(nums); // O(nlogn)

        for(int x = 0; x < nums.length; x++){
            if(x > 0 && nums[x] == nums[x-1]){
                continue;
            }

            int left = x+1;
            int right = nums.length - 1;

            while(left < right){

                int threeSum = nums[x] + nums[left] + nums[right];

                if(threeSum > 0){
                    right--;
                }
                else if(threeSum < 0){
                    left++;
                }
                else{
                    res.add(Arrays.asList(nums[x], nums[left], nums[right]));
                    left++;
                    right--;
                    while(left < right && nums[left] == nums[left-1]){
                        left++;
                    }
                }
            }

        }
        return res;

    }
}

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        l, r = 0, len(s)-1
        res = s[l:r+1]
        
        while l<r:
            if s[r] != s[l]:
                res = ""
                while l<r and s[r] != s[l]:
                    r-=1
                    l+=1
                res = s[l:r+1]

            r-=1
            l+=1
        return res